access/email/config.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	_ "embed"
+
+	"github.com/gravitational/teleport-plugins/lib"
+	"github.com/gravitational/teleport-plugins/lib/logger"
+	"github.com/gravitational/trace"
+	"github.com/pelletier/go-toml"
+)
+
+// DeliveryConfig represents email recipients config
+type DeliveryConfig struct {
+	Sender     string
+	Recipients []string
+}
+
+// MailgunConfig holds Mailgun-specific configuration options.
+type MailgunConfig struct {
+	Domain     string
+	PrivateKey string `toml:"private_key"`
+}
+
+// SMTPConfig is SMTP-specific configuration options
+type SMTPConfig struct {
+	Host     string
+	Port     int
+	Username string
+	Password string
+}
+
+// Config stores the full configuration for the teleport-email plugin to run.
+type Config struct {
+	Teleport lib.TeleportConfig `toml:"teleport"`
+	Mailgun  *MailgunConfig     `toml:"mailgun"`
+	SMTP     *SMTPConfig        `toml:"smtp"`
+	Delivery DeliveryConfig     `toml:"delivery"`
+	Log      logger.Config      `toml:"log"`
+}
+
+const exampleConfig = `# Example email plugin configuration TOML file
+
+[teleport]
+auth_server = "0.0.0.0:3025"                              # Teleport Auth Server GRPC API address
+
+# tctl auth sign --format=file --auth-server=auth.example.com:3025 --user=access-plugin --out=auth --ttl=1h
+identity = "/var/lib/teleport/plugins/email/auth"         # Teleport certificate ("file" format)
+
+# tctl auth sign --format=tls --auth-server=auth.example.com:3025 --user=access-plugin --out=auth --ttl=1h
+# client_key = "/var/lib/teleport/plugins/email/auth.key" # Teleport GRPC client secret key ("tls" format")
+# client_crt = "/var/lib/teleport/plugins/email/auth.crt" # Teleport GRPC client certificate ("tls" format")
+# root_cas = "/var/lib/teleport/plugins/email/auth.cas"   # Teleport cluster CA certs ("tls" format")
```

Comment:
From #295.

```suggestion
[teleport]
# Teleport Auth/Proxy Server address.
#
# Should be port 3025 for Auth Server and 3080 or 443 for Proxy.
# For Teleport Cloud, should be in the form "your-account.teleport.sh:443".
addr = "example.com:3025"

# Credentials.
#
# When using --format=file:
# identity = "/var/lib/teleport/plugins/email/auth_id"    # Identity file
#
# When using --format=tls:
# client_key = "/var/lib/teleport/plugins/email/auth.key" # Teleport TLS secret key
# client_crt = "/var/lib/teleport/plugins/email/auth.crt" # Teleport TLS certificate
# root_cas = "/var/lib/teleport/plugins/email/auth.cas"   # Teleport CA certs

```

Reply:
Done. Will replace auth_server with addr once base PR will be merged.